airflow/dag_processing/processor.py

```diff
@@ -735,6 +735,8 @@ def execute_callbacks_without_dag(
 
     @provide_session
     def _execute_dag_callbacks(self, dagbag: DagBag, request: DagCallbackRequest, session: Session):
+        from airflow.models.dag import DAG
```

Comment:
We should just move the `DAG` import out to global, the `dag` module is already loaded globally anyway.

Reply:
That was my first thought but just in case I decide to use local import here.

I will move out out TYPE_CHECKING block